### PR TITLE
Added support for night vision as an effect

### PR DESCRIPTION
**Pull request by alkemann**
#### Summary
```SUMMARY: Mods "Added support for night vision as effect"```

#### Purpose of change
More JSON support is more fun!

#### Describe the solution
To support mods like Magiclysm with the ability to through spells or other means give the player night vision, I added a check against the effect id "night_vision".

#### Describe alternatives you've considered
Considered adding the actual effect to core, but put it in the Magiclysm mod. Main has no issue with the check against the effect type when it doesn't exist.

#### Additional context
Also added a spell to Magiclysm mod that gives night vision called Dark sight. This spell may not be part of merge, depending on if @KorGgenT thinks it fits.

### CODE REVIEW

File: data/mods/Magiclysm/Spells/classless.json [hunk 1]
```diff
@@ -1,4 +1,22 @@
 [
+  {
+    "id": "nightvision",
+    "type": "SPELL",
+    "name": "Dark Sight",
+    "description": "Gives you the power to see in the dark",
+    "message": "Your eyes glow green for a second and suddenly your sight can pierce the darkest shadows.",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/Spells/classless.json]:
```suggestion
    "message": "Your eyes glow green for a moment. Now your sight can pierce the darkest shadows.",
```

File: data/mods/Magiclysm/Spells/classless.json [hunk 2]
```diff
@@ -1,4 +1,22 @@
...
+    "valid_targets": [ "self" ],
+    "effect": "target_attack",
+    "effect_str": "night_vision",
+    "min_range": 1,
+    "max_range": 1,
+    "base_casting_time": 100,
+    "base_energy_cost": 100,
```
Comment (KorGgenT) [inline data/mods/Magiclysm/Spells/classless.json]:
```suggestion
    "base_energy_cost": 400,
```
i prefer to err on the side of "a bit too weak" but i think 400 should be a good number for the length

File: data/mods/Magiclysm/effects/effects.json [hunk 3]
```diff
@@ -9,6 +9,15 @@
     "rating": "good",
     "base_mods": { "speed_mod": [ 20 ], "stamina_min": [ 10 ] }
   },
+  {
+    "type": "effect_type",
+    "id": "night_vision",
+    "name": [ "Dark Sight" ],
+    "desc": [ "You can see in the dark." ],
+    "apply_message": "Your sight adjust to the darkness.",
+    "remove_message": "The darkness looses it's shape.",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/effects/effects.json]:
```suggestion
    "remove_message": "The darkness loses it's shape.",
```

File: data/mods/Magiclysm/effects/effects.json [hunk 4]
```diff
@@ -9,6 +9,15 @@
     "rating": "good",
     "base_mods": { "speed_mod": [ 20 ], "stamina_min": [ 10 ] }
   },
+  {
+    "type": "effect_type",
+    "id": "night_vision",
+    "name": [ "Dark Sight" ],
+    "desc": [ "You can see in the dark." ],
+    "apply_message": "Your sight adjust to the darkness.",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/effects/effects.json]:
```suggestion
    "apply_message": "Your sight adjusts to the darkness.",
```

File: data/mods/Magiclysm/Spells/classless.json [hunk 5]
```diff
@@ -1,4 +1,22 @@
 [
+  {
+    "id": "dark_sight",
+    "type": "SPELL",
+    "name": "Dark Sight",
+    "description": "Gives you the power to see in the dark",
+    "message": "Your eyes glow green for a moment. Now your sight can pierce the darkest shadows.",
```
Comment (AMurkin) [inline data/mods/Magiclysm/Spells/classless.json]:
```suggestion
    "message": "Your eyes glow green for a moment.  Now your sight can pierce the darkest shadows.",
```

File: data/mods/Magiclysm/items/spell_scrolls.json [hunk 6]
```diff
@@ -10,6 +10,14 @@
     "symbol": "?",
     "color": "white"
   },
+  {
+    "type": "GENERIC",
+    "copy-from": "spell_scroll",
+    "id": "spell_scroll_dark_sight",
+    "name": "Scroll of Dark Sight",
+    "description": "The darknes holds no secrets for the arcane. Adjust your sight to see in perfect darkness!",
```
Comment (AMurkin) [inline data/mods/Magiclysm/items/spell_scrolls.json]:
```suggestion
    "description": "The darknes holds no secrets for the arcane.  Adjust your sight to see in perfect darkness!",
```

Comment (Night-Pryanik) [inline data/mods/Magiclysm/items/spell_scrolls.json]:
```suggestion
    "description": "The darkness holds no secrets for the arcane.  Adjust your sight to see in perfect darkness!",
```

File: data/mods/Magiclysm/Spells/classless.json (hunk 5 above)
Comment (alkemann) [reply data/mods/Magiclysm/Spells/classless.json]: Why two spaces?

Comment (AMurkin) [reply data/mods/Magiclysm/Spells/classless.json]: That's what we use for the in-game text: [Manual of style](/CleverRaven/Cataclysm-DDA/blob/master/doc/MANUAL_OF_STYLE.md).

Comment (alkemann) [reply data/mods/Magiclysm/Spells/classless.json]: hm. I guess there is no point in restating my question of why? :)

Comment (BevapDin) [reply data/mods/Magiclysm/Spells/classless.json]: @alkemann https://github.com/CleverRaven/Cataclysm-DDA/pull/5414#issuecomment-31611671

**Pull request merged**